### PR TITLE
[Dev16 Preview2]Single File Management Refactor

### PR DESCRIPTION
**Pull request by TIHan**
This PR depends on this one getting merged first: https://github.com/Microsoft/visualfsharp/pull/5833

This changes how we handle single files/script files. Also gets rid of the use of deprecated APIs. It's much simpler and resolves the issue when script files were re-named that all tooling would stop working for the script.

### CODE REVIEW

**Comment by TIHan** [comment]
@KevinRansom @cartermp this is ready assuming CI passes

**Comment by TIHan** [comment]
@cartermp @KevinRansom @brettfo This is ready. Nothing should regress and it should have our rename file issue resolved. There is more work to be done, however, it's important we make these changes in order to get rid of deprecated code that we are using from Roslyn.

**Comment by cartermp** [comment]
@TIHan and I did some extensive testing of this and think that the overall experience is good, plus it allows us to get rid of some rather fragile stuff on our end.

File: vsintegration/src/FSharp.Editor/LanguageService/FSharpProjectOptionsManager.fs [hunk 1]
```diff
@@ -305,45 +344,20 @@ type internal FSharpProjectOptionsManager
...
-                    return Some(parsingOptions, None, projectOptions)
-                | _ ->
-                    return None
+                return None
         }
 
     /// Get the options for a document or project relevant for syntax processing.
     /// Quicker then TryGetOptionsForDocumentOrProject as it doesn't need to recompute the exact project options for a script.
     member this.TryGetOptionsForEditingDocumentOrProject(document:Document) = 
-        let projectId = document.Project.Id
```
Comment (dsyme) [inline vsintegration/src/FSharp.Editor/LanguageService/FSharpProjectOptionsManager.fs:339]: @TIHan I don't understand this change in dev16.  

https://github.com/dotnet/fsharp/issues/10156 is a synchronous call to this on the UI thread

As the comment to this method says, the point of "TryGetOptionsForEditingDocumentOrProject" is to get editing options quickly for scripts, in the cases where we need them fast on the UI thread for editing operations.   But this change removed this and now this method computed the exact options.

I believe this is a primary cause for slower script editing in dev16, e.g. https://github.com/dotnet/fsharp/issues/6646, and is related to the UI thread aspects of https://github.com/dotnet/fsharp/issues/8757

Comment (TIHan) [reply vsintegration/src/FSharp.Editor/LanguageService/FSharpProjectOptionsManager.fs:339]: Looking back, I don't think I understood this well enough at the time and made the change in hopes to normalize everything.

If this call was being invoked on the UI thread somewhere, I can totally understand why there would be a performance issue.

**Pull request merged**